tests/filecheck/projects/riscv-backend-paper/bottom_up_f32.mlir

```diff
@@ -91,17 +91,13 @@ func.func public @pooling_nchw_max_d1_s2_3x3(
 // CHECK-NEXT:      mv t2, zero
 // CHECK-NEXT:      # Constant folded riscv_cf.bge
 // CHECK-NEXT:  scf_body_{{\d}}_for:
-// CHECK-NEXT:      li a2, 2
-// CHECK-NEXT:      mul a2, t2, a2
+// CHECK-NEXT:      slli a2, t2, 1
 // CHECK-NEXT:      li t6, 18
 // CHECK-NEXT:      mul a2, a2, t6
-// CHECK-NEXT:      li t6, 8
-// CHECK-NEXT:      mul a2, a2, t6                               # multiply by element size
+// CHECK-NEXT:      slli a2, a2, 3                               # multiply by element size
 // CHECK-NEXT:      add a2, t1, a2
-// CHECK-NEXT:      li t6, 8
-// CHECK-NEXT:      mul t6, t2, t6
-// CHECK-NEXT:      li t5, 8
-// CHECK-NEXT:      mul t6, t6, t5                               # multiply by element size
```

Comment:
These changes are scary and make me think that we might not want to actually add this change. The main issue is that other rewrites, like the one you proposed in the readme, rely on checking that an input is a product, and the check will fail here as it'll be a shift instead. I have a feeling that this optimisation should be in a different pass that optimises the code, as opposed to "canonicalization", which is designed to make the code more uniform to make downstream transformations easier.